src/common/mfu_daos.c

```diff
@@ -2556,38 +2575,160 @@ static int cont_serialize_props(struct hdf5_args *hdf5,
                                 daos_handle_t cont)
 {
     int                     rc = 0;
-    daos_prop_t             *prop_query;
-    uint32_t                i;
-    char                    cont_str[64];
-    int                     len = 0;
+    daos_prop_t*            prop_query;
+    daos_prop_t*            prop_acl = NULL;
+    struct daos_prop_entry* entry;
+    char                    cont_str[DAOS_PROP_LABEL_MAX_LEN];
 
     /*
      * Get all props except the ACL first.
      */
-    prop_query = daos_prop_alloc(1);
-    if (prop_query == NULL)
+    prop_query = daos_prop_alloc(DAOS_PROP_CO_NUM - 1);
+    if (prop_query == NULL) {
         return ENOMEM;
+    }
+
+    prop_query->dpp_entries[0].dpe_type = DAOS_PROP_CO_LABEL;
+    prop_query->dpp_entries[1].dpe_type = DAOS_PROP_CO_LAYOUT_TYPE;
+    prop_query->dpp_entries[2].dpe_type = DAOS_PROP_CO_LAYOUT_VER;
+    prop_query->dpp_entries[3].dpe_type = DAOS_PROP_CO_CSUM;
+    prop_query->dpp_entries[4].dpe_type = DAOS_PROP_CO_CSUM_CHUNK_SIZE;
+    prop_query->dpp_entries[5].dpe_type = DAOS_PROP_CO_CSUM_SERVER_VERIFY;
+    prop_query->dpp_entries[6].dpe_type = DAOS_PROP_CO_REDUN_FAC;
+    prop_query->dpp_entries[7].dpe_type = DAOS_PROP_CO_REDUN_LVL;
+    prop_query->dpp_entries[8].dpe_type = DAOS_PROP_CO_SNAPSHOT_MAX;
+    prop_query->dpp_entries[9].dpe_type = DAOS_PROP_CO_COMPRESS;
+    prop_query->dpp_entries[10].dpe_type = DAOS_PROP_CO_ENCRYPT;
+    prop_query->dpp_entries[11].dpe_type = DAOS_PROP_CO_OWNER;
+    prop_query->dpp_entries[12].dpe_type = DAOS_PROP_CO_OWNER_GROUP;
+    prop_query->dpp_entries[13].dpe_type = DAOS_PROP_CO_DEDUP;
+    prop_query->dpp_entries[14].dpe_type = DAOS_PROP_CO_DEDUP_THRESHOLD;
```

Comment:
@dsikich I chose to "hardcode" the indexes so it is more explicit when referenced below, since we can't use `daos_prop_entry_get`

Reply:
@daltonbohning yeah, that makes sense since we cannot use those.